### PR TITLE
Improve skins launch

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Launch ImproveDigital Collective skins for all. Capture all metrics when the auction is won by Prebid’s `improvedigital` bidder.

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## What is the value of this and can you measure success?

Better, faster, more valuable skins. More reporting on the skins.

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Tested

- [X] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/header-bidding/utils.ts [hunk 1]
```diff
@@ -142,12 +140,11 @@ export const shouldIncludeXaxis = (): boolean => isInUk();
...
-	return (
-		window.guardian.config.page.isFront &&
-		getBreakpointKey() === 'D' &&
-		isInABTestSynchronous(improveSkins)
-	);
+	return [
+		window.guardian.config.page.isFront,
+		isInUk(),
+		getBreakpointKey() === 'D',
+	].every(Boolean);
```
Comment (MarSavar) [inline static/src/javascripts/projects/commercial/modules/header-bidding/utils.ts:147]: I was curious about this pattern and ran a few tests in the console - it turns out that using `&&` to compare 3 elements is over 3 times faster than initialising an array and then running `Boolean` over each element. 
The `&&` operator took 3.57 seconds whereas `[...].every(Boolean)` took 11.1 seconds over 1 million iterations. If this code runs every time there's a Prebid auction it's probably worth using `&&`! 
What do you think?

**Review by marsavar — APPROVED** [review]
👍 !!

File: static/src/javascripts/projects/commercial/modules/header-bidding/utils.ts (hunk 1 above)
Comment (mxdvl) [reply static/src/javascripts/projects/commercial/modules/header-bidding/utils.ts:147]: Ha, I was actually wondering about the performance of this. Thanks for checking! I’ll revert back.

Do you mind sharing the benchmark?

Comment (MarSavar) [reply static/src/javascripts/projects/commercial/modules/header-bidding/utils.ts]: Sorry, late reply! Here's the screenshot: 
image

Comment (mxdvl) [reply static/src/javascripts/projects/commercial/modules/header-bidding/utils.ts]: Deno is (way) faster, but there’s still a 8× speedup.

```ts
const iterations = 1_000_000;

function test(id: string, fun: () => void) {
  console.time(id);
  fun();
  console.timeEnd(id);
}

test("every(Boolean)", () => {
  for (let index = 0; index < iterations; index++) {
    const _ = [true, true, true].every(Boolean);
  }
});

test("straight conditions", () => {
  for (let index = 0; index < iterations; index++) {
    const _ = true && true && true;
  }
});

```

**Comment by mxdvl** [comment]
This is now ready to go, @Amouzle & @ashishpuliyel!

File: static/src/javascripts/projects/commercial/modules/header-bidding/utils.ts [hunk 2]
```diff
@@ -142,11 +142,11 @@ export const shouldIncludeXaxis = (): boolean => isInUk();
 
 export const shouldIncludeImproveDigital = (): boolean => isInUk() || isInRow();
 export const shouldIncludeImproveDigitalSkin = (): boolean => {
-	if (!(isInUk() || isInRow())) return false;
 	return (
+		isInVariantSynchronous(improveSkins, 'variant') &&
```
Comment (ashishpuliyel) [inline static/src/javascripts/projects/commercial/modules/header-bidding/utils.ts:146]: @mxdvl need to measure both control and variant

File: static/src/javascripts/projects/commercial/commercial-metrics.ts [hunk 3]
```diff
@@ -25,17 +24,20 @@ const init = (): Promise<void> => {
...
 
 	return Promise.resolve();
 };
 
-export { init };
+/**
+ * A way to override the metrics sampling.
+ * It is safe to call this method repeatedly, as per [the `EventListener` docs](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#multiple_identical_event_listeners)
+ */
+const captureCommercialMetrics = (): void => {
```
Comment (ashishpuliyel) [inline static/src/javascripts/projects/commercial/commercial-metrics.ts:39]: This is nice and I like it. I would like something similar on core-web-vitals. @zekehuntergreen What do you think? Something we can call at any point in the page lifecycle. As I understand it vitals are only sent through onpageunload anyway, so if in the time the page has been loaded, if anyone said they care about recording the vitals, it'll record the vitals.

**Review by ashishpuliyel — CHANGES_REQUESTED** [review]
Overall looks good. But we need to record the metrics for the control as well as the sample, on request from D&I

File: static/src/javascripts/projects/commercial/modules/header-bidding/utils.ts (hunk 2 above)
Comment (mxdvl) [reply static/src/javascripts/projects/commercial/modules/header-bidding/utils.ts:146]: This ensures only the variant gets the placement id. Commercial metrics are sent for all variants.

File: static/src/javascripts/projects/commercial/commercial-metrics.ts [hunk 4]
```diff
@@ -2,21 +2,20 @@ import type { ABTest } from '@guardian/ab-core';
...
+	if (logged) return;
+	logged = sendCommercialMetrics(pageViewId, browserId, isDev);
+};
 
 const init = (): Promise<void> => {
-	if (!config.get('switches.commercialMetrics', false))
+	if (!window.guardian.config.switches.commercialMetrics)
 		return Promise.resolve();
 
 	const testsToForceMetricsFor: ABTest[] = [commercialPartner, improveSkins];
```
Comment (mxdvl) [inline static/src/javascripts/projects/commercial/commercial-metrics.ts:22]: @ashishpuliyel this is how we capture metrics for tests.

Can be tested by adding #ab-ImproveSkins=control

**Review by mxdvl — COMMENTED** [review]
I'm not sure we actually need the changes?

File: static/src/javascripts/projects/commercial/modules/header-bidding/utils.ts (hunk 2 above)
Comment (ashishpuliyel) [reply static/src/javascripts/projects/commercial/modules/header-bidding/utils.ts:146]: Got it, my mistake. Go for it then

**Review by ashishpuliyel — APPROVED** [review]

**Pull request merged**